src/components/overlays/dialog/Dialog.spec.ts

```diff
@@ -0,0 +1,81 @@
+import { describe, expect, it } from 'vitest';
+import {
+  type ComponentMountingOptions,
+  DOMWrapper,
+  mount,
+} from '@vue/test-utils';
+import Dialog from '@/components/overlays/dialog/Dialog.vue';
+import Button from '@/components/buttons/button/Button.vue';
+
+const createWrapper = (options: ComponentMountingOptions<typeof Dialog>) =>
+  mount(Dialog, { ...options, global: { stubs: { 'rui-button': Button } } });
+
+const delay = (time: number = 100) =>
+  new Promise((resolve) => {
+    setTimeout(resolve, time);
+  });
```

Comment:
nitpick: but we can use `promiseTimeout` from `@vueuse/core`

Reply:
oh yes, updating...